osbuild/util/ostree.py

```diff
@@ -225,16 +231,17 @@ def parse_deployment_option(root, deployment):
             raise ValueError("Could not find deployment")
         if len(filenames) > 1:
             raise ValueError("More than one deployment found")
-        
-        path = filenames[0].split("/")
-        osname = path[-3]
-        ref, serial = path[-1].split(".")
-        return osname, ref, serial
+
+        f = re.search("/ostree/deploy/(.*)/deploy/(.*)\\.([0-9])", filenames[0])
+        osname = f.group(1)
+        commit = f.group(2)
+        serial = f.group(3)
+        return osname, commit, serial
```

Comment:
Let's maybe spell it out a bit more for anyone coming along that will
wonder why the subtle behavior/policy here.

```suggestion
        # We pick up the osname commit and serial from the filesystem
        # here. We'll return the detected commit as the ref in this
        # since it's a valid substitute for all subsequent uses in
        # the code base.
        f = re.search("/ostree/deploy/(.*)/deploy/(.*)\\.([0-9])", filenames[0])
        osname = f.group(1)
        commit = f.group(2)
        serial = f.group(3)
        return osname, commit, serial
```